### PR TITLE
added postRequest method

### PR DESCRIPTION
**Pull request by Aras14HD**

### CODE REVIEW

File: src/HabiticaAPIManager.js [hunk 1]
```diff
@@ -177,6 +177,41 @@ class HabiticaAPIManager {
...
+	 * Make a POST request to the Habitica API.
+	 * Data object returned varies based on the API url called.
+	 * @param {string} baseURL - the url of the api call.
+	 * @param {HabiticaUserID} userID - the ID of the user, needed for authentication.
+	 * @param {HabiticaAPIToken} userAPIToken - the API Token for the user, needed for authentication.
+	 * @param {object} [queryParams={}] - key-value pairs for any parameters needed by the api call.
+	 * @returns {Promise<String>} Promise containing the raw API response data as a string.
+	 */
+	postRequest(baseURL, userID, userAPIToken, queryParams={}) {
+		let url = this.getQueryStringURL(baseURL, queryParams);
```
Comment (delightedCrow) [inline src/HabiticaAPIManager.js]: Thanks for the PR :)

For a POST request the parameters should go in the body of the request, not in the query string.

**Review by delightedCrow — CHANGES_REQUESTED** [review]
Have you tried testing your POST request against any of the HabiticaAPI's endpoints?

**Comment by Aras14HD** [comment]
I wanna add a createTask() function. Where should I put it? I was thinking either HabiticaAPIManager or HabiticaUser.

File: src/HabiticaAPIManager.js (hunk 1 above)
Comment (Aras14HD) [reply src/HabiticaAPIManager.js]: its resolved now!

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
 /node_modules
 .DS_Store
 /package-lock.json
+index.html
+index.js
+/docs
```
Comment (delightedCrow) [inline .gitignore]: The /docs folder needs to remain in the repository as it contains the project's demo code and reference documentation. It's also the source that GithHub Pages uses to host the demo and API docs for the project.

File: README.md [hunk 3]
```diff
@@ -12,10 +12,9 @@ For the most comprehensive and up-to-date documentation on HabiticaMagicJS funct
 
 ## Limitations
 
-HabiticaMagicJS currently supports only `GET` requests at the moment, so it works best for projects that just need **read-only access to the Habitica API**.
+HabiticaMagicJS currently supports only `GET` and `POST` requests at the moment, so it works best for projects that just need **read-only access to the Habitica API**.
```
Comment (delightedCrow) [inline README.md:15]: This wording is confusing, as the addition of POST request functionality negates the limitation of read-only access. 

I'd suggest either removing the Limitations section or at least changing this sentence to better describe which functionality is still limited or in progress.

File: src/HabiticaAPIManager.js [hunk 4]
```diff
@@ -177,6 +177,40 @@ class HabiticaAPIManager {
...
+		let promise = new Promise((resolve, reject) => {
+			let req = new XMLHttpRequest();
+			req.open("POST", baseURL);
+
+			req.onerror = function() {
+				reject(this.responseText);
+			};
+
+			req.onload = function() {
+				if (req.status === 201) {
```
Comment (delightedCrow) [inline src/HabiticaAPIManager.js]: Unlike with their GET request endpoints, the Habitica API endpoints for POST requests can return a few different success codes (like 200 and 201), so you'll want to make sure your generic postRequest function can handle all of the success codes the API returns.

**Review by delightedCrow — CHANGES_REQUESTED** [review]
This is an excellent start. 

We do need to keep the /docs directory though - I'd recommend just modifying the `README.md` and `src/HabiticaAPIManager.js` files for this PR.

**Pull request merged**